test/client/doc.js

```diff
@@ -475,4 +475,19 @@ describe('Doc', function() {
       ], done);
     });
   });
+
+  describe('snapshot generation', function() {
+    var doc;
+    beforeEach(function(done) {
+      doc = this.connection.get('dogs', 'scooby');
+      doc.create({name: 'Scooby'}, function(error) {
+        if (error) return done(error);
+        doc.whenNothingPending(done);
+      });
```

Comment:
```suggestion
      doc.create({name: 'Scooby'}, done);
```

We can probably simplify this setup — just less stuff to think about in future when looking at this test.